handshake_message_server_key_exchange.go

```diff
@@ -53,23 +53,29 @@ func (h *handshakeMessageServerKeyExchange) Unmarshal(data []byte) error {
 
 	publicKeyLength := int(data[3])
 	offset := 4 + publicKeyLength
-	if len(data) <= publicKeyLength {
```

Comment:
I approved (so feel free to merge away!)

Style wise do you think it is better to do one assert for total length `if len(data) <= handshakeMessageServerKeyExchangeLen` or small asserts before each data access?

Reply:
I think it is better to check as early as we can.
Probably we can use something like https://godoc.org/golang.org/x/crypto/cryptobyte to decrease probability of error like TLS library does.